graphql_compiler/schema_transformation/utils.py

```diff
@@ -17,10 +19,41 @@ class SchemaStructureError(SchemaTransformError):
     """
 
 
+class InvalidNameError(SchemaTransformError):
+    """Raised if a type/field name is not valid.
+
+    This may be raised if the input schema contains invalid names, or if the user attempts to
+    rename a type/field to an invalid name. A name is considered valid if it consists of
+    alphanumeric characters and underscores and doesn't start with a numeric character (as
+    required by GraphQL), and doesn't start with double underscores as such types are reserved
+    for GraphQL internal use.
+    """
+
+
 class SchemaNameConflictError(SchemaTransformError):
     """Raised when renaming types or fields cause name conflicts."""
 
 
+def check_name_is_valid(name):
```

Comment:
nit: `check_type_name_is_valid`, to be extra clear what kind of name we're talking about?